examples/text_to_image/train_text_to_image.py

```diff
@@ -0,0 +1,608 @@
+import argparse
+import copy
+import logging
+import math
+import os
+import random
+from pathlib import Path
+from typing import Optional
+
+import numpy as np
+import torch
+import torch.nn.functional as F
+import torch.utils.checkpoint
+
+from accelerate import Accelerator
+from accelerate.logging import get_logger
+from accelerate.utils import set_seed
+from datasets import load_dataset
+from diffusers import AutoencoderKL, DDPMScheduler, PNDMScheduler, StableDiffusionPipeline, UNet2DConditionModel
+from diffusers.optimization import get_scheduler
+from diffusers.pipelines.stable_diffusion import StableDiffusionSafetyChecker
+from huggingface_hub import HfFolder, Repository, whoami
+from torchvision import transforms
+from tqdm.auto import tqdm
+from transformers import CLIPFeatureExtractor, CLIPTextModel, CLIPTokenizer
+
+
+logger = get_logger(__name__)
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(description="Simple example of a training script.")
+    parser.add_argument(
+        "--pretrained_model_name_or_path",
+        type=str,
+        default=None,
+        required=True,
+        help="Path to pretrained model or model identifier from huggingface.co/models.",
+    )
+    parser.add_argument(
+        "--dataset_name",
+        type=str,
+        default=None,
+        help=(
+            "The name of the Dataset (from the HuggingFace hub) to train on (could be your own, possibly private,"
+            " dataset)."
+        ),
+    )
+    parser.add_argument(
+        "--dataset_config_name",
+        type=str,
+        default=None,
+        help="The config of the Dataset, leave as None if there's only one config.",
+    )
+    parser.add_argument("--train_data_dir", type=str, default=None, help="A folder containing the training data.")
```

Comment:
Add something like:

```
Folder contents must follow the structure described in https://huggingface.co/docs/datasets/image_dataset#imagefolder. In particular, a `metadata.jsonl` file must exist to provide the captions for the images.
```

Also, a comment about this option being ignored if `--dataset_name` is present.

Reply:
Actually I'm not sure it's worthwhile to provide both options. Could we do with just local or remote Datasets?

Reply:
Think it's important to have both options. Some users may not want to share their dataset on the hub and would like to use the local options, while some other users may want to play with existing datasets on the hub. It's not complicated to support both.

And thanks a lot for the suggestions, changed the description accordingly.

Reply:
I didn't mean to only support remote datasets, but use the same arg to load either remote datasets (using their name) or local ones (using a path). AFAIK both are supported by `Datasets`, so that's why I saw no need for an additional argument. But I may be missing some detail.